internal/metrics/otelmetrics/factory.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package otelmetrics
+
+import (
+	"context"
+	"log"
+
+	"go.opentelemetry.io/otel/attribute"
+	metric "go.opentelemetry.io/otel/metric"
+
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+)
+
+type otelFactory struct {
+	meter metric.Meter
+}
+
+func NewFactory(meterProvider metric.MeterProvider) metrics.Factory {
+	return &otelFactory{
+		meter: meterProvider.Meter("jaeger-v2"),
+	}
+}
+
+func (f *otelFactory) Counter(opts metrics.Options) metrics.Counter {
+	counter, err := f.meter.Int64Counter(opts.Name)
+	if err != nil {
+		log.Printf("Error creating OTEL counter: %v", err)
+		return metrics.NullCounter
+	}
+
+	attributeSet := f.CreateAtrributeSet(opts.Tags)
+	return &otelCounter{
+		counter:  counter,
+		fixedCtx: context.Background(),
+		option:   metric.WithAttributeSet(attributeSet),
+	}
+}
+
+func (f *otelFactory) Gauge(opts metrics.Options) metrics.Gauge {
+	gauge, err := f.meter.Int64Gauge(opts.Name)
+	if err != nil {
+		log.Printf("Error creating OTEL gauge: %v", err)
+		return metrics.NullGauge
+
+	}
+
+	attributeSet := f.CreateAtrributeSet(opts.Tags)
+	return &otelGauge{
+		gauge:    gauge,
+		fixedCtx: context.Background(),
+		option:   metric.WithAttributeSet(attributeSet),
+	}
+}
+
+func (f *otelFactory) Histogram(opts metrics.HistogramOptions) metrics.Histogram {
+	histogram, err := f.meter.Float64Histogram(opts.Name)
+	if err != nil {
+		log.Printf("Error creating OTEL histogram: %v", err)
+		return metrics.NullHistogram
+	}
+
+	attributeSet := f.CreateAtrributeSet(opts.Tags)
+	return &otelHistogram{
+		histogram: histogram,
+		fixedCtx:  context.Background(),
+		option:    metric.WithAttributeSet(attributeSet),
+	}
+}
+
+func (f *otelFactory) Timer(opts metrics.TimerOptions) metrics.Timer {
+	timer, err := f.meter.Float64Histogram(opts.Name)
+	if err != nil {
+		log.Printf("Error creating OTEL timer: %v", err)
+		return metrics.NullTimer
+	}
+	attributeSet := f.CreateAtrributeSet(opts.Tags)
+	return &otelTimer{
+		histogram: timer,
+		fixedCtx:  context.Background(),
+		option:    metric.WithAttributeSet(attributeSet),
+	}
+}
+
+func (f *otelFactory) Namespace(opts metrics.NSOptions) metrics.Factory {
+	return f
+}
+
+func (f *otelFactory) CreateAtrributeSet(tags map[string]string) attribute.Set {
```

Comment:
Receiver `f` is not required. Can be private function.
```suggestion
func attributeSetOption(tags map[string]string) metric.Option {
```